### ISSUE TITLE
Slideshow block: improve accessibility

### ISSUE THREAD

**Issue by simison** [Automattic/wp-calypso#30897]
via p1HpG7-6jw-p2 — by @davidakennedy 

### Structure

- [x] **The markup for the slides could be improved.** Right now, the slides are marked up as `div` elements, which provide no semantic meaning to screen readers and other assistive technology. The slides could be marked up as an unordered list, that way we get the number of slides announced for free to assistive technology.

### Functionality

- [x] **The markup for the previous/next buttons could be improved.** Right now, these work well, but they’re marked up as `div` elements. It would be better to just make them `button` elements. That way, they get all the built-in functionality that comes with native buttons. If that’s changed, the `tabindex` set to 0 won’t be needed.
- [x] **The markup for the navigation buttons could be improved.** The same could be said for the pagination dots. They technically could be buttons, instead of `div` elments since they perform an action.
- [x] **The current slide should be announced.** (⚠️ High Priority) This can be done with ARIA live regions. WordPress has an API to make this easier. Check out [these](https://foxland.fi/wordpress-speaks-wp-a11y-speak/) [posts](https://make.wordpress.org/accessibility/2015/04/15/let-wordpress-speak-new-in-wordpress-4-2/). I see some `aria-live` information in the HTML source, but it doesn’t seem to announce anything. Perhaps we need to enable that in some way?
- [x] **Overall, focus states need to be added.** (⚠️ High Priority) Many of the interactive elements, the buttons and slides themselves need stronger focus states that don’t rely on color alone.
- [x] **Manage focus.** (⚠️ High Priority) When the navigation buttons are clicked, we should move focus to the slide that’s selected. We shouldn’t do this with the previous/next buttons though, because it will make it harder for some to move forward and back.
- [x] **Using it with a screen reader, I can’t actually move into the individual slides.** (⚠️ High Priority)

### Animations

- [x] **When the slideshow is autoplayed, it should have a pause button** (⚠️ High Priority) This is a key requirement for people who may be sensitive to motion on a page. The slides can autoplay, but a button to pause it affords control to those who need it.
- [x] **Pausing the slideshow on hover/focus:** We could also improve things by pausing the autoplaying slideshow on hover and focus of a slide. This is usually more important if the slides have links or text information, but this would be a nice addition.

### Styling

- [x] **Showing current slide with shape.** (⚠️ High Priority) It’s good to show the status of the slides in the navigation beyond just color changes. Changing the circle shape to square for the current slide would be a good change here.
- [x] **Contrast on focus styles need to be better.** (⚠️ High Priority) There aren’t any focus on the buttons or slides as far as I can tell. There should be, and they should meet minimum contrast requirements.

**Comment by jeffersonrabb** [comment]
All of these issues were addressed and resolved in https://github.com/Automattic/wp-calypso/pull/30895. They were originally raised and discussed in 
`p1HpG7-6jw-p2`.
